From 8aec9e4c89002c30f67cc02dbd9d8a608a56beae Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:05:20 -0400
Subject: [powerpc] Enable /dev/port when isa_io_special is set

Message-id: <1410545655-205645-92-git-send-email-dzickus@redhat.com>
Patchwork-id: 94123
O-Subject: [RHEL7 PATCH 091/626] powerpc: Enable /dev/port when isa_io_special is set
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit ac237b65f56c9b80d7774c35ccce15a74d445621
Author: Benjamin Herrenschmidt <benh@kernel.crashing.org>
Date:   Thu Aug 29 16:55:07 2013 +1000

    powerpc: Enable /dev/port when isa_io_special is set

    isa_io_special is set when the platform provides a "special"
    implementation of inX/outX via some FW interface for example.

    Such a platform doesn't need an ISA bridge on PCI, and so /dev/port
    should be made available even if one isn't present.

    This makes the LPC bus IOs accessible via /dev/port on PowerNV Power8

    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/include/asm/io.h b/arch/powerpc/include/asm/io.h
index db1f296..575fbf8 100644
--- a/arch/powerpc/include/asm/io.h
+++ b/arch/powerpc/include/asm/io.h
@@ -21,7 +21,7 @@ extern struct pci_dev *isa_bridge_pcidev;
 /*
  * has legacy ISA devices ?
  */
-#define arch_has_dev_port() (isa_bridge_pcidev != NULL)
+#define arch_has_dev_port() (isa_bridge_pcidev != NULL || isa_io_special)
 #endif
 
 #include <linux/device.h>
-- 
1.7.1